File: Recruitment.API/Recruitment.API/Models/Notification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Recruitment.API.Models
{
    public class Notification
    {
        [Key]
        public int id { get; set; }
        public int userId { get; set; }

        [ForeignKey("userId")]
        public virtual User user { get; set; }

        public string title { get; set; } = string.Empty;
        public string content { get; set; } = string.Empty;
        public bool isRead { get; set; } = false;
        public DateTime createDate { get; set; } = DateTime.Now;

        // --- CÁC TRƯỜNG MỚI ĐỂ ĐIỀU HƯỚNG ---
        // Phân loại thông báo: "FOLLOW", "APPLY", "COMMENT", "REVIEW", "SYSTEM"
        public string type { get; set; } = string.Empty;

        // ID tham chiếu: Có thể là ID của Follower, ID của Application, hoặc ID của Job
        public int? referenceId { get; set; }
    }
}

File: Recruitment.UI/src/types/notification.ts
export enum NotificationType {
    COMMENT = 'COMMENT',
    REPLY = 'REPLY',
    REVIEW = 'REVIEW',
    NEW_JOB = 'NEW_JOB',
    FOLLOW = 'FOLLOW',
    UNFOLLOW = 'UNFOLLOW',
}

export interface Notification {
    id: number;
    userId: number;
    title: string;
    content: string;
    isRead: boolean;
    createDate: string;
    type?: NotificationType;     // COMMENT | REVIEW | NEW_JOB | FOLLOW | UNFOLLOW
    referenceId?: number;        // userId (FOLLOW/UNFOLLOW) hoặc jobId (COMMENT/REVIEW/NEW_JOB)
}

export interface NotificationState {
    notifications: Notification[];
    unreadCount: number;
    loading: boolean;
    error: string | null;
}